src/App.tsx

```diff
@@ -1,7 +1,37 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressedKey: string | null;
+};
+
+export class App extends React.Component<State> {
+  state = {
+    pressedKey: null,
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ pressedKey: event.key });
+    });
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ pressedKey: event.key });
+    });
+  }
```

Comment:
Functions are objects in JavaScript, and objects are stored by reference. It means that your callback function in the 13th row isn't the same as the one in the 19th row, as a result, this event listener won't be removed.
To fix it, you should simply create a function and pass it as a callback in both methods `addEventListener` and `removeEventListener`.

https://javascript.info/object-copy
https://javascript.info/introduction-browser-events#addeventlistener

![image](https://user-images.githubusercontent.com/102293862/232526305-a51abbd2-53c0-4917-b617-a168fbd76f84.png)